### PR TITLE
TST: Avoid time dependency in GCS zip test

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

To avoid CI flakiness from this test, only compare the checksum of the ZIP contents to avoid comparing the flaky last modified timestamp of the ZIP archive. IMO, the CI stability is worth the the trade off of being a little less strict here.

cc @jbrockmendel

### CODE REVIEW

**Comment by jbrockmendel** [comment]
this is fantastic

**Review by jbrockmendel — APPROVED** [review]

File: pandas/tests/io/test_gcs.py [hunk 1]
```diff
@@ -88,16 +89,22 @@ def test_to_read_gcs(gcs_buffer, format):
...
+    to avoid checking the time-dependent last-modified timestamp which
+    in some CI builds is off-by-one
 
     See https://en.wikipedia.org/wiki/ZIP_(file_format)#File_headers
     """
-    assert result[:9] == expected[:9]
-    assert result[11:] == expected[11:]
+    if compression == "zip":
+        # Only compare the CRC checksum of the file contents
+        res = zipfile.ZipFile(BytesIO(result))
```
Comment (jreback) [inline pandas/tests/io/test_gcs.py:102]: don't you need to do these in with clauses?

**Review by jreback — CHANGES_REQUESTED** [review]
can you rebase as well

File: pandas/tests/io/test_gcs.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/io/test_gcs.py]: Fair point. Changed.

**Comment by jbrockmendel** [comment]
LGTM

**Pull request merged**